File: filter/filter-names.js
const readline = require('readline');
const fs = require('fs');
let names = [];

const rl = readline.createInterface({
    input: fs.createReadStream('../data/names.txt')
});

rl.on('line', (line) => {
    names.push(line);
});

rl.on('close', () => {
    console.time('start');
    names = names.filter(name => name.indexOf('A') === -1);
    console.timeEnd('start');
    console.log(names.length);
});

File: filter/filter-names.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"time"
)

func Filter(vs []string, f func(string) bool) []string {
	vsf := vs[:0]
	for _, v := range vs {
		if f(v) {
			vsf = append(vsf, v)
		}
	}
	return vsf
}

func main() {
	file, err := os.Open("../data/names.txt")
	if err != nil {
		panic("error in opening the file")
	}

	names := []string{}
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		names = append(names, scanner.Text())
	}

	fmt.Println("count is", len(names))
	start := time.Now()

	names = Filter(names, func(name string) bool { return !strings.Contains(name, "A") })

	elapsed := time.Since(start)
	fmt.Println(elapsed)

	fmt.Println("count is", len(names))
}
